rust/lance/src/dataset/transaction.rs

```diff
@@ -1381,6 +1385,10 @@ impl Transaction {
             )
         };
 
+        if let Some(commit_msg) = &config.commit_message {
+            manifest.update_config(vec![(COMMIT_MESSAGE_KEY.to_string(), commit_msg.clone())]);
+        }
+
```

Comment:
Another point to think about is that, this could be considered really as a composite transaction of a write + a ConfigUpdate. If we finish the transaction v2 work, that could potentially achieve the same result.